### PR TITLE
Credentials

### PR DESCRIPTION
**Pull request by jamestalton**

### CODE REVIEW

**Review by vidyanambiar — COMMENTED** [review]
Looks good. 🎉 
I see a few pending items (Filtering by credential type, type details section in the create/edit form for different credential types and RBAC checks for enabling the delete button) but those can be addressed in subsequent PRs.

**Comment by vidyanambiar** [comment]
I see the CredentialDetail view also implemented in this PR from @AlexSCorey https://github.com/ansible/ansible-ui/pull/201

File: frontend/awx/resources/credentials/CredentialForm.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import { useTranslation } from 'react-i18next';
```
[the remark this answers is not in the document]
Comment (jamestalton) [reply frontend/awx/resources/credentials/CredentialForm.tsx:1]: This should be addressed with the changed merged in from main.

File: frontend/awx/resources/credentials/CredentialForm.tsx [hunk 2]
```diff
@@ -22,19 +23,21 @@ import { getAwxError } from '../../useAwxView';
...
     try {
-      try {
-        const organization = await getOrganizationByName(
-          editedCredential.summary_fields.organization.name
-        );
-        if (!organization) throw new Error(t('Organization not found.'));
-        editedCredential.organization = organization.id;
-      } catch {
-        throw new Error(t('Organization not found.'));
+      if (credential.summary_fields.organization.name) {
```
Comment (vidyanambiar) [inline frontend/awx/resources/credentials/CredentialForm.tsx:28]: If the org is not provided we should provide the current user in the payload:
https://github.com/ansible/awx/blob/devel/awx/ui/src/screens/Credential/CredentialAdd/CredentialAdd.js#L61

Comment (jamestalton) [reply frontend/awx/resources/credentials/CredentialForm.tsx:29]: done

**Review by vidyanambiar — APPROVED** [review]
lgtm!
We'll need to add RBAC for the create/delete buttons in a future PR.

File: cypress/e2e/awx/resources/credentials.cy.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    cy.typeByLabel(/^Name$/, 'a');
+    cy.clickButton(/^Save credential$/);
+    cy.hasTitle(`${credential.name}a`);
+  });
+
+  it('credential details', () => {
+    cy.clickRow(credential.name);
+    cy.hasTitle(credential.name);
+    cy.clickButton(/^Details$/);
+    cy.contains('#name', credential.name);
```
Comment (vidyanambiar) [inline cypress/e2e/awx/resources/credentials.cy.ts:64]: Further to the discussion on assertions in the aap-ui channel, we'll need explicit assertions in some of these tests. 
eg: `cy.contains('#name', credential.name).should('exist');`

**Pull request merged**